agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -618,3 +618,197 @@ func TestNetworkModeBridge(t *testing.T) {
 	err := networkModeTest(t, agent, "bridge")
 	require.NoError(t, err, "Networking mode bridge testing failed")
 }
+
+// TestCustomAttributesWithValidOptions tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with a key, value pair
+func TestCustomAttributesWithValidOptions(t *testing.T) {
+	len := 25
+	key, value := GetRandomString(len), GetRandomString(len)
+
+	customAttribute := fmt.Sprintf("{\"%s\": \"%s\"}", key, value)
+	os.Setenv("ECS_INSTANCE_ATTRIBUTES", customAttribute)
+
+	agent := RunAgent(t, nil)
+	defer agent.Cleanup()
+
+	params := &ecsapi.DescribeContainerInstancesInput{
+		Cluster:            &agent.Cluster,
+		ContainerInstances: []*string{&agent.ContainerInstanceArn},
+	}
+
+	resp, err := ECS.DescribeContainerInstances(params)
+	require.NoError(t, err)
+	require.NotEmpty(t, resp.ContainerInstances)
```

Comment:
[multiple occurrences] can you also do a `require.Len(1, resp.ContainerInstances, ..)` here?